cou/steps/analyze.py

```diff
@@ -53,6 +53,7 @@ async def create(cls, model_name: Optional[str] = None) -> Analysis:
         apps = await Analysis._populate(model_name)
 
         control_plane, data_plane = cls._split_control_plane_and_data_plane(apps)
+        control_plane, data_plane = cls._remove_data_plane_apps(control_plane, data_plane)
```

Comment:
These two functions are both returning control_plane and data_plane apps and are not being used anywhere else, I think we should merge them.

Reply:
I do not agree with this. There is clearly a separate logic, so both deserve their own function.

Reply:
I agree with Erhan here.

Reply:
I think we can have a middle term by calling ` _move_to_data_plane_if_required` inside `_split_control_plane_and_data_plane` and return control_plane and data_plane